mozilla/blurts-server
# Refresh stored scans before sending activity email

# References: 
Jira: MNTOR-3196
Figma: N/A




# Description

I also moved the call to `markMonthlyActivityEmailAsJustSent` to just before sending the email, rather than before fetching the data. After all, if something goes wrong fetch the data but we haven't sent an email yet, it's no problem to retry sending it later.

(I also renamed `profileId` to `onerepProfileId`, just to make sure I didn't pass `subscriber.id` in there.)

# How to test

Run `npm run dev:cronjobs` with a user that still expects a monthly email (helpful query: `UPDATE subscribers SET monthly_monitor_report_at = NULL WHERE monthly_monitor_report_at IS NOT NULL`). It should work without problems. I also added some logs to `refreshStoredScanResults` to show that it actually fetched data from OneRep.

# Checklist (Definition of Done)
- [ ] I've added a unit test to test for potential regressions of this bug. - N/A, only DB interaction.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage. (Marked as unactionable.)
